### DISCUSSION TITLE
System wide notifications with proper Slack integration

### DISCUSSION THREAD

**Discussion by bimtauer** [flyteorg/flyte#3647] [RFC Incubator]
Flyte currently only allows sending notifications by email and requires users to explicitly define notification rules in their launchplans. 

A better notifications setup would enable
- configuring system-wide alerts (no need to explicitly enable notifications on a per-launchplan basis)
- limiting alerts to a subset of available states, i.e. receive notification for all workflow failures and aborts
- limit for certain domains/projects/workflows
- message templating for Slack

As an inspiration I would point to [ArgoCD Notifications](https://argocd-notifications.readthedocs.io/en/stable/), I think they implemented a very nice solution with triggers, templates and subscriptions.

**Comment by davidmirror-ops** [comment]
@bimtauer Sorry for the delay
Watch [here](https://youtu.be/lSTGi6noyas?t=2392) the discussion around this idea. The general consensus is that it's a great one and it fits in the scope of an RFC

See the RFC process [here](https://github.com/flyteorg/flyte/tree/master/rfc#rfc-process) and let us know any question/comment

Thank you!

**Comment by davidmirror-ops** [comment]
2023-11-09 Contributor's meetup notes: this entry needs a champion to make it an RFC. It's dependent on the development of Webhooks: https://github.com/flyteorg/flyte/pull/4147

**Comment by davidmirror-ops** [comment]
@bimtauer closing this as you can, starting with `flytekit >=1.15.1`, use the [Webhook Agent](https://github.com/flyteorg/flytekit/pull/3058) to integrate with Slack via webhooks and push notifications. Any further question please let us know
